retrieval/target.go

```diff
@@ -267,6 +266,30 @@ func (t *Target) String() string {
 	return t.host()
 }
 
+// fingerprint returns an identifying hash for the target.
+func (t *Target) fingerprint() model.Fingerprint {
+	t.RLock()
+	defer t.RUnlock()
+
+	return t.labels.Fingerprint()
+}
+
+// offset returns the time until the next scrape cycle for the target.
+func (t *Target) offset(interval time.Duration) time.Duration {
+	now := time.Now().UnixNano()
+
+	var (
+		base   = now % int64(interval)
```

Comment:
To be equivalent to before, you needed `-(now % int64(interval)` here.

Or you check in L287 if `next > interval` (and then substract interval). The latter might be most intuitive.